tests/functional/tests/smoke.py

```diff
@@ -0,0 +1,132 @@
+            assert Path("/snap/bin/cou").exists()
+            cls.exc_path = "/snap/bin/cou"
+
+        else:
+            # functest already installs cou as python package, but we install again
+            # for better developer experience
+            log.warning("using cou as python package")
+            assert check_call(f"python3 -m pip install {os.getenv('PYTHONPATH')}".split()) == 0
+            cls.exc_path = "cou"
+
+    def remove_snap_package() -> None:
+        """Remove cou package."""
+        if os.environ.get("TEST_SNAP"):
+            log.info("Removing snap package cou")
+            check_call("sudo snap remove charmed-openstack-upgrader --purge".split())
+
+    def cou(self, cmd: list[str]) -> str:
+        """Run cou commands.
+
+        :param cmd: Command to run.
+        :type cmd: list[str]
+        :return: Response of the command with the stderr on stdout.
+        :rtype: str
+        """
+        try:
+            return check_output([self.exc_path] + cmd, stderr=STDOUT).decode()
+        except CalledProcessError as err:
+            log.error(err.output)
+
+    def test_plan_backup(self) -> None:
+        """Test plan with backup."""
+        result = self.cou(["plan", "--model", self.model_name])
+        expected_plan = (
+            "Upgrade cloud from 'ussuri' to 'victoria'\n"
+            "\tVerify that all OpenStack applications are in idle state\n"
+            "\tBackup mysql databases\n"
+            "\tControl Plane principal(s) upgrade plan\n"
+            "\t\tUpgrade plan for 'designate-bind' to victoria\n"
+            "\t\t\tUpgrade software packages of 'designate-bind' "
+            "from the current APT repositories\n"
+            "\t\t\tUpgrade 'designate-bind' to the new channel: 'victoria/stable'\n"
+            "\t\t\tWait 300s for app designate-bind to reach the idle state.\n"
+            "\t\t\tCheck if the workload of 'designate-bind' has been upgraded\n"
+        )
+        self.assertIn(expected_plan, result)
+
+    def test_plan_no_backup(self) -> None:
+        """Test plan with no backup."""
+        result = self.cou(["plan", "--model", self.model_name, "--no-backup"])
+        expected_plan = (
+            "Upgrade cloud from 'ussuri' to 'victoria'\n"
+            "\tVerify that all OpenStack applications are in idle state\n"
+            "\tControl Plane principal(s) upgrade plan\n"
+            "\t\tUpgrade plan for 'designate-bind' to victoria\n"
+            "\t\t\tUpgrade software packages of 'designate-bind' "
+            "from the current APT repositories\n"
+            "\t\t\tUpgrade 'designate-bind' to the new channel: 'victoria/stable'\n"
+            "\t\t\tWait 300s for app designate-bind to reach the idle state.\n"
+            "\t\t\tCheck if the workload of 'designate-bind' has been upgraded\n"
+        )
+        self.assertIn(expected_plan, result)
+
+    def test_plan_no_backup_choosing_model(self) -> None:
```

Comment:
Since we need to always passing in model name, this test is no longer needed (duplicate of `test_plan_no_backup`)